src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -836,22 +837,28 @@ public Response findUsers(@Context final HttpServletRequest httpServletRequest,
                         }
                     }
                 }
-                findUsers = subjectFilteredUsers;
+                foundUsers = subjectFilteredUsers;
             }
 
             // Calculate the ETag
-            EntityTag etag = new EntityTag(findUsers.size() + findUsers.toString().hashCode()
+            EntityTag etag = new EntityTag(foundUsers.size() + foundUsers.toString().hashCode()
                     + userPrototype.toString().hashCode() + "");
 
             Response cachedResponse = generateCachedResponse(request, etag);
             if (cachedResponse != null) {
                 return cachedResponse;
             }
 
+            if (foundUsers.size() > 2000) {
```

Comment:
Notifying the users on the front end when 2000 results have been returned would be a good idea.